aux_process.c

```diff
@@ -0,0 +1,1537 @@
+
+            /* if no chars left in this token, move to next token */
+            if (argv[aux_optind][++sp] == '\0') {
+                aux_optind++;
+                sp = 1;
+            }
+            return '?';
+        }
+
+        if (*++cp == ':') {
+            /* if a value is expected, get it */
+            if (argv[aux_optind][sp + 1] != '\0') {
+                /* flag value is rest of current token */
+                aux_optarg = &argv[aux_optind++][sp + 1];
+            }
+            else if (++aux_optind >= argc) {
+                if (aux_opterr)
+                    fprintf(stderr, "%s: value expected for option \"%c\"\n", argv[0], optchar);
+
+                optchar = '?';
+            }
+            else {
+                /* flag value is next token */
+                aux_optarg = argv[aux_optind++];
+            }
+
+            sp = 1;
+        }
+        /* wildcard argument */
+        else if (*cp == '*') {
+            /* check the next argument */
+            aux_optind++;
+            /* we do have an extra argument, check if not last */
+            if ((aux_optind + 1) < argc) {
+                if (argv[aux_optind][0] != '-') {
+                    aux_optarg = argv[aux_optind++];
+                }
+                else {
+                    aux_optarg = NULL;
+                }
+            }
+            else {
+                aux_optarg = NULL;
+            }
+        }
+        else {
+            /* set up to look at next char in token, next time */
+            if (argv[aux_optind][++sp] == '\0') {
+                /* no more in current token, so setup next token */
+                aux_optind++;
+                sp = 1;
+            }
+            aux_optarg = NULL;
+        }
+    }
+
+    /* return the current flag character found */
+    return optchar;
+}
+
+/*-------------------------------------------------------------------------
+ * Function: do_sleep
+ *
+ * Purpose:  Sleep a time of TICK_LEN / POLLS_PER_TICK
+ *
+ * Return:   Success:    0
+ *
+ *           Failure:    -1
+ *-------------------------------------------------------------------------
+ */
+static int
+do_sleep(handler_t *hand)
+{   
+    struct timespec time, time2;
+
+    time.tv_sec = (hand->tick_len / hand->polls_per_tick) / 10;
+    time.tv_nsec = (hand->tick_len * 100 * 1000 * 1000 / hand->polls_per_tick) % (1000 * 1000 * 1000);
+    
+    if (nanosleep(&time, &time2) < 0) {
+        fprintf(stderr, "Nano sleep system call failed \n");
+        return -1;
+    }
+
+    return 0; 
+}       
+
+/*-------------------------------------------------------------------------
+ * Function: usage
+ *
+ * Purpose:  Show command usage
+ *-------------------------------------------------------------------------
+ */
+static void
+usage(void)
+{
+    printf("    [-h] [-a --skip_aux] [-c --vfd_config] [-l --log_file_path] [-m --md_chksum_path] [-p --polls_per_tick] [-s --stats] [-t --tick_len] [-v --verbose]\n"); 
```

Comment:
For the command line, please add <md_file> and <ud_path> after the options so that user will know to specify those two files as they are needed.   Also add description for <md_file> and <ud_path>.

Reply:
Done.